PyInstaller/hooks/hook-numpy.py

```diff
@@ -1,54 +1,159 @@
+                names.add(path.name)
+    if not binaries:
+        logger.warning(
+            "Failed to find '%s' DLL in PATH. Your app will likely crash if run"
+            " on a different machine that doesn't already have it.", name
         )
-    extend_system_path(numpy_core_paths)
-    del numpy_core_paths
-
-# if we bundle the testing module, this will cause
-# `scipy` to be pulled in unintentionally but numpy imports
-# numpy.testing at the top level for historical reasons.
-# excludedimports = collect_submodules('numpy.testing')
-
-binaries = []
-
-# package the DLL bundle that official numpy wheels for Windows ship
-# The DLL bundle will either be in extra-dll on windows proper
-# and in .libs if installed on a virtualenv created from MinGW (Git-Bash
-# for example)
-if is_win:
-    extra_dll_locations = ['extra-dll', '.libs']
-    for location in extra_dll_locations:
-        dll_glob = os.path.join(os.path.dirname(
-            get_module_file_attribute('numpy')), location, "*.dll")
-        if glob.glob(dll_glob):
-            binaries.append((dll_glob, "."))
+    return binaries
+
+
+if is_mkl:
+    # Other dlls that MKL uses which PyInstaller can't detect itself. These are
+    # determined empirically using dynamic dependency sniffing and will lead
+    # to issues in the future should they change...
+    [binaries.extend(find_library(i))
+     for i in ["libcrypto*", "libffi*", "libssl*"]]
+
+
+# --- A vain attempt at Conda's numpy support ---
+
+# Regular numpy, even with unofficial mkl builds, is pretty trivial to support
+# with PyInstaller. Unfortunately Conda's numpy is the opposite. We need
+# their help to maintain this because our own attempts have been a disaster.
+
+if compat.is_conda:
+    hiddenimports.append("six")
+
+    # There are so many hidden binary dependencies. This list is heavily, OS,
+    # Python and NumPy versions dependent. Omitting any of these can lead to
+    # obscure and often traceback-less crashes.
+    # XXX: As you can see, this is really not a scalable solution. Needs help!
+    conda_dll_patterns = [re.compile(i) for i in (
+        'apphelp.*', 'crypt32.*', 'imagehlp.*', 'libblas.*',
+        'libcblas.*', 'libcrypto.*', 'libffi.*', 'libgcc_.*',
+        'libgfortran.*', 'libifcoremd.*', r'libiomp\d+md.*', 'liblapack.*',
+        'libmmd.*', 'libomp.*',
+        'libopenblas.*', 'libquadmath.*', 'libssl.*', 'libuuid.*',
+        'libz.*', 'mkl_avx.*', 'mkl_core.*',
+        'mkl_intel_thread.*', 'mkl_rt.*', 'mkl_vml_avx.*',
+        'mkl_vml_avx.*', 'msasn.*', 'mswsock.*', 'ole.*',
+        'oleaut.*', 'tbbmalloc.*', 'urandom'
+    )]
+
+    if compat.is_win:
+        lib_dir = os.path.join(compat.base_prefix, "Library", "bin")
+    else:
+        lib_dir = os.path.join(compat.base_prefix, "lib")
+
+    _to_add = set()
+    [
+        _to_add.add(name) for name in os.listdir(lib_dir)
+        if any(pattern.match(name) for pattern in conda_dll_patterns)
+    ]
```

Comment:
```suggestion
    _to_add = set(
        filter(
            lambda x: any(p.match(x) for p in conda_dll_patterns),
            os.listdir(lib_dir)
        )
    )
```
You might want to check this actually runs before committing it if you do 😅.